src/braket/circuits/noise_helpers.py

```
@@ -62,10 +62,11 @@ def check_noise_target_gates(noise: Noise, target_gates: Iterable[Type[Gate]]):
 
     if noise.qubit_count > 1:
         for g in target_gates:
-            if g != Gate.Unitary and g().qubit_count != noise.qubit_count:
+            fixed_qubit_count = g.fixed_qubit_count()
+            if g != Gate.Unitary and fixed_qubit_count != noise.qubit_count:
```

If Unitary is an input, should we raise an error?